strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/PyPiArtifactCoordinates.java

```diff
@@ -0,0 +1,168 @@
+package org.carlspring.strongbox.artifact.coordinates;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.semver.Version;
```

Comment:
Question: Is PyPi strictly semver-based, or does it also accept free-style versions?

Reply:
I've never seen non-semver packages IRL, but the possibility is supported - https://www.python.org/dev/peps/pep-0440/#version-scheme

I'm afraid covering all of those variants would be not practical. We could make this data driven and look up how popular those naming  conventions are on PyPi...

Reply:
The problem is that not everybody follows the norms. For example, either Maven, nor NPM, or NuGet are strict on this. In theory,  both NPM and NuGet should be using Semver, but in real life, if you have a look at the packages available on these repositories, you'll see that a large part of the people aren't adhering to these rules. If we lock down the version validation to just Semver by default, this is likely to cause problems for many people. Also, could you please have a look at our page on [[Artifact Coordinae Validators]]?

Reply:
With all possible version schemes and frankly nothing holding a person from making up their own, version format is frankly not parseable. Do we just treat is as a string then?